lib/block_view_lockups.go

```diff
@@ -0,0 +1,2178 @@
+		bav.Handle, bav.Snapshot, hodlerPKID, profilePKID, currentTimestampNanoSecs)
+	if err != nil {
+		return nil, errors.Wrapf(err, "UtxoView.GetUnlockableLockedBalanceEntries")
+	}
+	for _, lockedBalanceEntry := range dbUnlockableLockedBalanceEntries {
+		// Cache results in the UtxoView.
+		if _, exists := bav.LockedBalanceEntryKeyToLockedBalanceEntry[lockedBalanceEntry.ToMapKey()]; !exists {
+			bav._setLockedBalanceEntry(lockedBalanceEntry)
+		}
+	}
+
+	// Then, pull unlockable LockedBalanceEntries from the UtxoView.
+	var unlockableLockedBalanceEntries []*LockedBalanceEntry
+	for _, lockedBalanceEntry := range bav.LockedBalanceEntryKeyToLockedBalanceEntry {
+		// Filter to matching LockedBalanceEntries.
+		if !lockedBalanceEntry.HODLerPKID.Eq(hodlerPKID) ||
+			!lockedBalanceEntry.ProfilePKID.Eq(profilePKID) ||
+			lockedBalanceEntry.UnlockTimestampNanoSecs > currentTimestampNanoSecs ||
+			lockedBalanceEntry.BalanceBaseUnits.IsZero() ||
+			lockedBalanceEntry.isDeleted {
+			continue
+		}
+		unlockableLockedBalanceEntries = append(unlockableLockedBalanceEntries, lockedBalanceEntry)
+	}
+
+	// Sort UnlockableLockedBalanceEntries by timestamp ASC.
+	sort.Slice(unlockableLockedBalanceEntries, func(ii, jj int) bool {
+		return unlockableLockedBalanceEntries[ii].UnlockTimestampNanoSecs <
+			unlockableLockedBalanceEntries[jj].UnlockTimestampNanoSecs
+	})
+	return unlockableLockedBalanceEntries, nil
+}
+
+//
+// TYPES: LockupYieldCurvePoint
+//
+
+type LockupYieldCurvePoint struct {
+	ProfilePKID               *PKID
+	LockupDurationNanoSecs    int64
+	LockupYieldAPYBasisPoints uint64
+	isDeleted                 bool
+}
+
+type LockupYieldCurvePointKey struct {
+	ProfilePKID            PKID
+	LockupDurationNanoSecs int64
+}
+
+func (lockupYieldCurvePoint *LockupYieldCurvePoint) Copy() *LockupYieldCurvePoint {
+	return &LockupYieldCurvePoint{
+		ProfilePKID:               lockupYieldCurvePoint.ProfilePKID.NewPKID(),
+		LockupDurationNanoSecs:    lockupYieldCurvePoint.LockupDurationNanoSecs,
+		LockupYieldAPYBasisPoints: lockupYieldCurvePoint.LockupYieldAPYBasisPoints,
+	}
+}
+
+func (lockupYieldCurvePoint *LockupYieldCurvePoint) Eq(other *LockupYieldCurvePoint) bool {
+	return lockupYieldCurvePoint.ToMapKey() == other.ToMapKey()
+}
+
+func (lockupYieldCurvePoint *LockupYieldCurvePoint) ToMapKey() LockupYieldCurvePointKey {
+	return LockupYieldCurvePointKey{
+		ProfilePKID:            *lockupYieldCurvePoint.ProfilePKID,
+		LockupDurationNanoSecs: lockupYieldCurvePoint.LockupDurationNanoSecs,
+	}
+}
+
+// DeSoEncoder Interface Implementation for LockupYieldCurvePoint
+
+func (lockupYieldCurvePoint *LockupYieldCurvePoint) RawEncodeWithoutMetadata(blockHeight uint64, skipMetadata ...bool) []byte {
+	var data []byte
+	data = append(data, EncodeToBytes(blockHeight, lockupYieldCurvePoint.ProfilePKID, skipMetadata...)...)
+	data = append(data, UintToBuf(uint64(lockupYieldCurvePoint.LockupDurationNanoSecs))...)
```

Comment:
Idk what happens if you call uint64(negativeInt64). Does it just work or does it panic?